cmd/srrs/main_test.go

```diff
@@ -109,25 +122,86 @@ func TestAll(t *testing.T) {
 	a.Equal(api.State{}, got)
 	log.Debug("Nil state received on WebSocket")
 
+	// state TRC -> SRRC
 	for i := 0; i < 10; i++ {
-		// TODO: generate
-		state := &api.State{
-			Turtles: map[string]*api.TurtleState{
-				"foo": {},
-			},
-		}
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			state := apitest.RandomState()
+
+			log.Debug("Sending random state TRC -> SRRC")
+			err = trc.SendState(state)
+			a.NoError(err)
+
+			got := &api.State{}
+			log.Debug("Waiting for state on WebSocket...")
+			err = wsConn.ReadJSON(got)
+			a.Nil(err)
+			a.Equal(got, state)
+			log.Debug("Random state received on WebSocket")
+		})
+	}
+
+	// state SRRC -> TRC
+	for i := 0; i < 10; i++ {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			state := apitest.RandomState()
+
+			log.Debug("Sending random state SRRC -> TRC")
+			err = wsConn.WriteJSON(state)
+			a.NoError(err)
+
+			got := &api.State{}
+			log.Debug("Waiting for state on TRC...")
+			msg := <-msgCh
+			a.Equal(msg.Type, api.MessageTypeState)
+			err := json.Unmarshal(msg.Payload, got)
+			a.Nil(err)
+			a.Equal(got, state)
+			log.Debug("Random state received on TRC")
+		})
+	}
 
-		log.Debug("Sending random state...")
-		err = trc.SendState(state)
-		a.NoError(err)
-		log.Debug("Random state sent")
-
-		got := &api.State{}
-		log.Debug("Receiving random state on WebSocket...")
-		err = wsConn.ReadJSON(got)
-		a.Nil(err)
-		a.Equal(got, state)
-		log.Debug("Random state received on WebSocket")
+	// command TRC -> SRRS
+	for i := 0; i < 10; i++ {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			c := apitest.RandomCommand()
+			state := &api.State{
+				Command: *c,
+			}
+
+			log.Debugf("Sending command %s", c)
+			err = trc.SendState(state)
+			a.NoError(err)
+			log.Debug("Random command sent")
+
+			got := &api.State{}
+			log.Debug("Receiving random state on WebSocket...")
+			err = wsConn.ReadJSON(got)
+			a.Nil(err)
+			a.Equal(got.Command, c)
+			log.Debug("Random state received on WebSocket")
+		})
+	}
+
+	// command SRRC -> TRC
+	for i := 0; i < 10; i++ {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			c := apitest.RandomCommand()
+			state := &api.State{
+				Command: *c,
+			}
+
+			log.Debug("Sending command %s SRRC -> TRC", c)
+			err = wsConn.WriteJSON(state)
```

Comment:
That's not how it works. WebSocket is read-only. States/commands are sent via REST API, see the spec on drive.